src/UnrolledUtilities.jl

```diff
@@ -56,22 +56,20 @@ include("generatively_unrolled_functions.jl")
     error("unrolled_applyat has detected an out-of-bounds index")
 
 @inline unrolled_reduce(op, itr, init) =
-    (rec_unroll(itr) ? rec_unrolled_reduce : gen_unrolled_reduce)(op, itr, init)
-@inline unrolled_reduce(op, itr; init = NoInit()) =
     isempty(itr) && init isa NoInit ?
     error("unrolled_reduce requires an init value for empty iterators") :
+    (rec_unroll(itr) ? rec_unrolled_reduce : gen_unrolled_reduce)(op, itr, init)
+@inline unrolled_reduce(op, itr; init = NoInit()) =
```

Comment:
Is this method needed? It looks like the positional arg-based method could just have an optional `init`.

Reply:
The kwarg version is intended to match `Base.reduce` (similar to `Base.mapreduce`, which can't have a positional arg method with an optional `init` because it takes a variable number of input iterators).

Reply:
Right, that's fine then.